### PR TITLE
chore: upgrade momento sdk - consistency, less copypasta

### PR DESCRIPTION
**Pull request by kvcache**
This change establishes a common pattern for interacting with the
service. Rather than copy/pasting matches, MomentoError -> CliError
mapping and json dump code around, this names those functionalities
as functions:
* interact_with_momento: Completes a momento future and maps the
  result to a Result< HappyCaseType, CliError >
* print_whatever_this_is_as_json: prints whatever you feed it as json

This change was precipitated by the change in the SDK to simplify
and coalesce client creation behind the SimpleCacheClientBuilder and
triggered by the SDK upgrade.

There are small changes to debug and error messages, but this change
should cause no functional changes.

tired of copypasta.
```
pub async fn delete_cache(cache_name: String, auth_token: String) -> Result<(), CliError> {
    debug!("deleting cache...");
    let mut momento = get_momento_client(auth_token).await?;
    match momento.delete_cache(&cache_name).await {
        Ok(_) => (),
        Err(e) => return Err(CliError { msg: e.to_string() }),
    };
    Ok(())
}
```
becomes
```
pub async fn delete_cache(cache_name: String, auth_token: String) -> Result<(), CliError> {
    let mut client = get_momento_client(auth_token).await?;
    interact_with_momento("deleting cache...", client.delete_cache(&cache_name)).await
}
```

### CODE REVIEW

File: src/utils/client.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+pub fn print_whatever_this_is_as_json<T>(value: &T)
+where
+    T: serde::Serialize,
+{
+    println!(
+        "{}",
+        serde_json::to_string_pretty(value).expect("Could not print whatever this is as json")
+    );
+}
```
Comment (tylerburdsall) [inline src/utils/client.rs:34]: Minor nit: I like having a function that uses generics to pretty print results, could we name it to something more streamlined like `print_as_json`? `print_whatever_this_is` almost seems a little tongue-in-cheek for a public-vended CLI

**Review by tylerburdsall — COMMENTED** [review]
Minor comment, everything else looks good. Thank you for helping reduce our copy-pasta code, this is super clean

File: src/utils/client.rs (hunk 1 above)
Comment (kvcache) [reply src/utils/client.rs:34]: Sure, I'll remove this tiny island of joy from the sea of dispassion. I still need to look into windows too.

**Review by cprice404 — APPROVED** [review]
nice cleanup, ty

**Comment by kvcache** [comment]
converted to draft because I need to mess with the github workflow to make windows work. I don't know why it can't find cmake.

**Comment by kvcache** [comment]
Integration tests are broken for unrelated reasons.

**Review by cprice404 — APPROVED** [review]

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Review by tylerburdsall — APPROVED** [review]

**Pull request merged**